### PR TITLE
chore: Centralise ESLint rules

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
A single `.eslintrc.js` to the entire repository.
This is a initial stage to normalize code standards and validate Linting rules for every PR.

FIXES CORE-504

### CODE REVIEW

**Comment by olivermrbl** [comment]
Really like the proposal!

QQ: A single `.eslintrc.js` for the entire repository forces developers to open the monorepo for linting to kick in right? You won't be able to open `packages/medusa` and still have it format correctly on save. It's not an issue but I imagine some developers open single packages for ease of navigation of files.

**Comment by carlos-r-l-rodrigues** [comment]
Yeah, considering it is a monorepo, all the configs would be centralized.
If that become an issue in the future we can have these configs as a package, and all other packages could have a minimal config just importing that

.eslintrc
```
{
  "extends": [
    "@medusajs/eslint-config"
  ],
}
```

**Review by olivermrbl — APPROVED** [review]
LGTM, before merging we should make sure @medusajs/engineering is on board with the changes 👍

File: .eslintrc.js [hunk 1]
```diff
@@ -1,77 +1,118 @@
...
   rules: {
-    "prettier/prettier": `error`,
-    curly: [2, `all`],
-    "new-cap": `off`,
-    "require-jsdoc": `off`,
-    semi: `off`,
-    "no-unused-expressions": `off`,
-    camelcase: `off`,
-    "no-invalid-this": `off`,
+    curly: [1, "all"],
```
[the remark this answers is not in the document]
Comment (carlos-r-l-rodrigues) [reply .eslintrc.js:13]: nice catch, that was a mistake.. I'll use the string to be explicit.

**Review by pKorsholm — APPROVED** [review]
Looks good!

**Review by adrien2p — APPROVED** [review]
LGTM !

**Review by kasperkristensen — APPROVED** [review]
LGTM!

**Pull request merged**